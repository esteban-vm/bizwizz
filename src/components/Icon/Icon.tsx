import type { FC, IconName } from '@/types'
import { IconNames } from './Icons'

interface IconProps {
  name: IconName
  className?: string
  onClick?: () => void
}

const Icon: FC<IconProps> = ({ name, ...props }) => {
  return (
    <svg role='img' {...props}>
      <use xlinkHref={`#icon-${IconNames[name]}`} />
    </svg>
  )
}

export default Icon
