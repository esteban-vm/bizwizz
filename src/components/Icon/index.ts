export { default as Icon } from './Icon'
export { default as Icons, IconNames } from './Icons'
