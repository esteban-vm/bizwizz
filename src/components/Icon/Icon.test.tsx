import { render, cleanup, screen } from '@/tests'
import Icon from './Icon'
import Icons from './Icons'

describe('Icon:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(
      <>
        <Icons />
        <Icon className='fill-primary' name='desktop' />
      </>
    ))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const iconElem = screen.getByRole('img', { hidden: true })
    expect(iconElem).toBeInTheDocument()
    expect(iconElem).toHaveAttribute('class', 'fill-primary')
  })
})
