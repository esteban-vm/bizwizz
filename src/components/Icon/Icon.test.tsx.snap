// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon: > should render correctly 1`] = `
<DocumentFragment>
  <svg
    class="fill-primary"
    role="img"
  >
    <use
      xlink:href="#icon-7"
    />
  </svg>
</DocumentFragment>
`;
